Project/Source/Components/ComponentMeshRenderer.cpp

```diff
@@ -976,3 +988,19 @@ void ComponentMeshRenderer::UpdateDissolveAnimation() {
 float ComponentMeshRenderer::GetDissolveValue() const {
 	return dissolveAnimationReverse ? 1.0f - dissolveThreshold : dissolveThreshold;
 }
+
+void ComponentMeshRenderer::UpdateTextureTiling(int programId, float2 value) const {
+	if (textureTiling[0] - 1.0f > 0.0001f || textureTiling[1] - 1.0f > 0.0001f) {
+		glUniform2fv(programId, 1, textureTiling.ptr());
+	} else {
+		glUniform2fv(programId, 1, value.ptr());
+	}
+}
+
+void ComponentMeshRenderer::UpdateTextureOffset(int programId, float2 value) const {
+	if (textureOffset[0] > 0.0001f || textureOffset[1] > 0.0001f) {
+		glUniform2fv(programId, 1, textureOffset.ptr());
+	} else {
+		glUniform2fv(programId, 1, value.ptr());
+	}
+}
```

Comment:
I think these two functions should have just a value as an input and have them return a value that is used to set the uniform in the Draw function, so you don't need to pass the programId. I don't know, it makes sense to me since we're setting the uniforms only in the Draw function and we just pass them values.